### PR TITLE
Android CircleCI to GitHub Actions reusable workflow conversion 

### PR DESCRIPTION
**Pull request by timkimadobe**
## Questions for reviewers
For the standardized build and test workflow:
1. Are the current make rule names desired as the standard?

> [!WARNING]
> This PR also makes a breaking change to the iOS build-and-test workflow to rename the iOS integration test command from `test-integration-upstream` -> `integration-test-ios` to align with the rule naming convention.

## Description

> [!NOTE]
> See #64 for additional context on reusable workflows
> This PR is the Android equivalent to the iOS workflows in #67 

This PR:
1. Converts CircleCI PR check workflow into GitHub Action workflow format
3. Is defined as a reusable workflow so that other repos are able to take advantage of the common flow, and are able to trigger on whatever criteria they need (the intended use is on PR creation)
4. Enables Codecov reporting on all test jobs

There are three new files:
### `.github/actions/android-setup-dependencies-action/action.yml`
* Configures Java and optimizes disk space by removing unnecessary tools to improve Android emulator stability.
* Adds KVM group permissions to enable efficient Android emulator execution with proper permissions.
* Sets up Android Virtual Device (AVD) caching to speed up emulator boot.
    * If `android-api-level` is not provided, skips this sequence

### `android-build-and-test.yml`
Build and test (CircleCI converted to GitHub actions reusable workflow) - which has the standard PR check make commands
* Uses new composite action which handles env setup and caching logic to reduce boilerplate: `.github/actions/android-setup-dependencies-action/action.yml`
* Adds job run flags so that callers can control which checks are run for their repo (each are optional, and default to `false`)
* Allows for matrix of Android API levels for testing
    * Uses defaults if matrix values are not provided by the caller

### `android-custom-command-build-and-test.yml`
This is a modularized workflow that handles: 
* Matrix logic - checking if matrix values were passed in or not, and providing default values
* Caching logic
* Running the make command

> [!NOTE]
> This modular workflow is used by `android-build-and-test.yml` to avoid repeating a lot of the same setup code, and it can also be called by repos directly if they need to use non-standard `make` commands.

Example caller repo usage:
```yaml
name: Build and Test

on:
  pull_request:

jobs:
  build-and-test:
    name: " "
    uses: adobe/aepsdk-commons/.github/workflows/android-build-and-test.yml@gha-android-3.0.0
    with:
      android-api-levels: '[29, 30]'
      run-test-unit: true
      run-test-functional: true
      run-test-integration: true
      run-build-app: true
    secrets: inherit
```

### Example runs

Example run (with matrix): https://github.com/timkimadobe/aepsdk-edge-android/actions/runs/11926866755
* How the PR checks look: https://github.com/timkimadobe/aepsdk-edge-android/pull/10
* Codecov results: https://app.codecov.io/github/timkimadobe/aepsdk-edge-android/commit/6df245ff8803ee569c5ee27c4168d60633657db8

## Motivation and Context

The goal is to reduce the amount of workflow maintenance work that needs to be done across repos (ex: action version upgrades, runner changes, iOS simulator updates, etc).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by praveek — APPROVED** [review]

**Comment by praveek** [comment]
Before merging remember to update the GH references from your fork to point to this repo?

File: .github/workflows/android-build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      run-test-integration:
+        type: boolean
+        default: false
+      run-build-app:
+        type: boolean
+        default: false
+      enable-codecov:
+        description: 'Enable Codecov for test coverage.'
+        type: boolean
+        default: true
```
Comment (addb) [inline .github/workflows/android-build-and-test.yml:41]: nit: either we default everything to false or think about flipping some other defaults to true like unit and functional which we usually want to run for every build.

File: .github/workflows/android-build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    uses: timkimadobe/aepsdk-commons/.github/workflows/android-custom-command-build-and-test.yml@android-circleci-to-gha
+    with:
+      android-api-levels: ${{ inputs.android-api-levels }}
+      command: make functional-test-coverage
+      enable-codecov: ${{ inputs.enable-codecov }}
+      codecov-flag: android-functional-tests
+    secrets: inherit
+
+  test-integration:
+    name: Integration Test (Android)
```
Comment (addb) [inline .github/workflows/android-build-and-test.yml:73]: nit: Do we need to prefix (Android) ?

File: .github/workflows/android-build-and-test.yml (hunk 1 above)
Comment (addb) [reply .github/workflows/android-build-and-test.yml:41]: Sounds good!

**Pull request merged**